src/Session.php

```diff
@@ -874,11 +874,17 @@ public static function checkValidSessionId()
 
         $user_id    = self::getLoginUserID();
         $profile_id = $_SESSION['glpiactiveprofile']['id'] ?? null;
-        $entity_id  = $_SESSION['glpiactive_entity'] ?? null;
+
+        $raw_entities = array_column($_SESSION['glpiprofiles'], 'entities');
+        $entities = [];
+        foreach ($raw_entities as $raw_entity) {
+            $entities = array_merge($entities, array_keys($raw_entity));
+        }
+        $entities = array_unique($entities);
```

Comment:
If a user has rights on 100 entities with a limited profile and in 1 entity with an admin profile, check on admin profile will be considered as valid for all the 101 entities.